flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -66,6 +68,7 @@
 	private Map<Map<String, String>, HiveTablePartition> partitionSpec2HiveTablePartition = new HashMap<>();
 	private boolean initAllPartitions;
 	private boolean partitionPruned;
+	private int[] projectedFields;
 
 	public HiveTableSource(JobConf jobConf, ObjectPath tablePath, CatalogTable catalogTable) {
 		this.jobConf = Preconditions.checkNotNull(jobConf);
```

Comment:
Invoke `this()` to init variable, like `projectedFields`, you can init it as `IntStream.range(0, rowArity).toArray()`

Reply:
I'd like to keep `projectedFields ` as null to indicate no projection has been pushed down to this table source. So that `explainSource()` doesn't have to display all the column indices if no projection is ever applied. What do you think?

Reply:
It is OK